dom/src/MainDOMSource.ts

```diff
@@ -111,28 +116,36 @@ export class MainDOMSource implements DOMSource {
     };
   }
 
-  public elements(): MemoryStream<Element> {
-    let output$: Stream<Element | Array<Element>>;
+  public elements(): MemoryStream<Array<Element>> {
+    let output$: Stream<Array<Element>>;
     if (this._namespace.length === 0) {
-      output$ = this._rootElement$;
+      output$ = this._rootElement$.map(x => [x]);
     } else {
       const elementFinder = new ElementFinder(
         this._namespace,
         this._isolateModule,
       );
       output$ = this._rootElement$.map(el => elementFinder.call(el));
     }
-    const out: DevToolEnabledSource & MemoryStream<Element> = adapt(
+    const out: DevToolEnabledSource & MemoryStream<Array<Element>> = adapt(
       output$.remember(),
     );
     out._isCycleSource = this._name;
     return out;
   }
 
+  public element(): MemoryStream<Element> {
+    return this.elements().map(arr => arr[0]);
```

Comment:
I propose that we filter this array to remove null values.

I feel that in most use cases, when you call `.element()` you are looking to perform operations using the data in that element. In these cases, you would end up filtering out null values.

I can see a use case for when you would want to get the nulls, so that you could tell when an element had yet to render or has been removed, but I don't think that's a common usecase right now, and you could get that information from `.elements()`.

We could always make an argument that defaults to removing the nulls if we wanted to support that use case.